main.go

```diff
@@ -192,6 +217,10 @@ func main() {
 
 	setupCreateOrderRoute(r, db)
 
+	setupSearchRoutes[User](r, db, "users")
```

Comment:
`setupSearchRoute`. it set's one route for the the specified entity.

may be we can also call it in the setupCRUDRoutes, so you need to add just one line instead of three.